tests/components/tplink_omada/conftest.py

```diff
@@ -53,29 +54,33 @@ def mock_setup_entry() -> Generator[AsyncMock]:
 
 
 @pytest.fixture
-def mock_omada_site_client() -> Generator[AsyncMock]:
+async def mock_omada_site_client(hass: HomeAssistant) -> AsyncGenerator[AsyncMock]:
     """Mock Omada site client."""
     site_client = MagicMock()
 
-    gateway_data = json.loads(load_fixture("gateway-TL-ER7212PC.json", DOMAIN))
```

Comment:
Why can't a sync test function not use `load_fixture`?
Do we force now all tests to be async if they want to load a fixture?

Reply:
> Why can't a sync test function not use `load_fixture`?
> Do we force now all tests to be async if they want to load a fixture?

`async def _get_mock_known_clients` was already async - so it popped up in the pylint checker.

As I migrated that one, I reviewed the other functions and felt it was better to migrate all of them to `async`.

Please note also that I am not touching any test functions - only regular functions and test fixtures